inlong-audit/audit-source/src/main/java/org/apache/inlong/audit/sink/pulsar/PulsarClientService.java

```diff
@@ -0,0 +1,252 @@
+        }
+        if (event.getHeaders().containsKey(AttributeConstants.INLONG_GROUP_ID)) {
+            groupId = event.getHeaders().get(AttributeConstants.INLONG_GROUP_ID);
+            proMap.put(AttributeConstants.INLONG_GROUP_ID, groupId);
+        }
+
+        logger.debug("producer send msg!");
+        producer.newMessage().properties(proMap).value(event.getBody())
+                .sendAsync().thenAccept((msgId) -> {
+            sendMessageCallBack.handleMessageSendSuccess((MessageIdImpl)msgId, es);
+
+        }).exceptionally((e) -> {
+            sendMessageCallBack.handleMessageSendException(es, e);
+            return null;
+        });
+        return true;
+    }
+
+    /**
+     * If this function is called successively without calling {@see #destroyConnection()}, only the
+     * first call has any effect.
+     *
+     * @throws FlumeException if an RPC client connection could not be opened
+     */
+    private void createConnection(CreatePulsarClientCallBack callBack) throws FlumeException {
+        if (pulsarClient != null) {
+            return;
+        }
+        try {
+            pulsarClient = initPulsarClient(pulsarServerUrl);
+            callBack.handleCreateClientSuccess(pulsarServerUrl);
+        } catch (PulsarClientException e) {
+            callBack.handleCreateClientException(pulsarServerUrl);
+            logger.error("create connnection error in metasink, "
+                            + "maybe pulsar master set error, please re-check.url{}, ex1 {}",
+                    pulsarServerUrl,
+                    e.getMessage());
+        } catch (Throwable e) {
+            callBack.handleCreateClientException(pulsarServerUrl);
+            logger.error("create connnection error in metasink, "
+                            + "maybe pulsar master set error/shutdown in progress, please "
+                            + "re-check. url{}, ex2 {}",
+                    pulsarServerUrl,
+                    e.getMessage());
+        }
+    }
+
+    private PulsarClient initPulsarClient(String pulsarUrl) throws Exception {
+        return PulsarClient.builder()
+                .serviceUrl(pulsarUrl)
+                .connectionTimeout(clientOpTimeout, TimeUnit.SECONDS)
+                .build();
+    }
+
+    public Producer initTopicProducer(String topic) {
+        logger.info("initTopicProducer topic = {}", topic);
+        Producer producer = null;
+        try {
+            producer = pulsarClient.newProducer().sendTimeout(sendTimeout,
+                    TimeUnit.MILLISECONDS)
+                    .topic(topic)
+                    .enableBatching(enableBatch)
+                    .blockIfQueueFull(blockIfQueueFull)
+                    .maxPendingMessages(maxPendingMessages)
+                    .batchingMaxMessages(maxBatchingMessages)
+                    .create();
+        } catch (PulsarClientException e) {
+            logger.error("create pulsar client has error e = {}", e);
```

Comment:
init Producer failed，throw exception is better.

Reply:
return null will be retry here.